packages/contracts-bedrock/src/L1/SystemConfig.sol

```diff
@@ -195,6 +201,14 @@ contract SystemConfig is OwnableUpgradeable, ISemver {
         _setStartBlock();
 
         _setResourceConfig(_config);
+
+        l2ChainId = _l2ChainId;
+    }
+
+    /// @notice Upgrades the SystemConfig by setting the L2 chain ID variable.
+    /// @param _l2ChainId The L2 chain ID that this SystemConfig configures.
+    function upgrade(uint256 _l2ChainId) external reinitializer(initVersion()) {
+        l2ChainId = _l2ChainId;
```

Comment:
Related to above comment, we can alteratively remove this input and just read it from the dispute games

Reply:
Would prefer to keep as-is for now. We can remove this input when we make the dispute games have shared inputs.